### PR TITLE
Feature: "Minimum Support Area" and "Minimum Support Interface Area"

### PR DESCRIPTION
**Pull request by vgribinchuk**
Feature: Minimum Support Area / Minimum Interface Area.
Backend implementation of the feature: https://github.com/Ultimaker/Cura/pull/4486

### CODE REVIEW

File: src/support.cpp [hunk 1]
```diff
@@ -1418,19 +1424,22 @@ void AreaSupport::generateSupportRoof(SliceDataStorage& storage, const SliceMesh
...
-    interface_polygons = support_areas.intersection(model);
+    Polygons interface_polygons = support_areas.intersection(model);
     interface_polygons = interface_polygons.offset(safety_offset).intersection(support_areas); //Make sure we don't generate any models that are not printable.
-    interface_polygons.removeSmallAreas(1.0);
+    if (minimum_interface_area > 0.0)
+    {
+        interface_polygons.removeSmallAreas(minimum_interface_area);
+    }
     support_areas = support_areas.difference(interface_polygons);
+    return interface_polygons;
```
Comment (nallath) [inline src/support.cpp]: This does increase the number of copies when compared to the previous situation. Returning the polygons (without reference) means that an additional copy is done.

Comment (vgribinchuk) [reply src/support.cpp]: Here is the place where C++ compiler is able to apply NRVO eliminating copying of the object, thus no additional overhead.

**Comment by nallath** [comment]
for our reference; CURA-5903

**Comment by rburema** [comment]
Hi @vgribinchuk, Same as for the frontend, some conflicts got introduced. Could you fix those? cheers, Remco.

**Pull request merged**